lib/linguist/generated.rb

```diff
@@ -113,6 +114,13 @@ def generated?
     def xcode_file?
       ['.nib', '.xcworkspacedata', '.xcuserstate'].include?(extname)
     end
+    
+    # Is the blob an IntelliJ IDEA file (IDE-specific files that are sometimes checked in)? 
+    #
+    # Returns true or false.
+    def intellij_file?
+      !!name.match(/.idea\//)
```

Comment:
```suggestion
      !!name.match(/(?:^|\/)\.idea\//)
```
The regex you have now is too permissive:
* Without an anchor like `^` or `\/` prefixing `.idea`, partial directory names will be matched; e.g., `./my.bright.idea/foo.txt`
* An unescaped `.` will match any single character in a a regular expression, so `/.idea/` will match `1idea/`, `files/Another idea/`, etc.

Reply:
Damn, you're right. Was a bit too hasty with this one. 😅